### PR TITLE
Don't recompile XEH EH's before every call

### PR DESCRIPTION
**Pull request by dedmen**
Medical XEH_init is always recompiled before being used
Before
First spawn: slack_2017-11-04_18-13-39
Second spawn: arma3_2017-11-04_18-14-53

After:
First spawn: 
Second spawn: arma3_2017-11-04_18-53-49

I'm checking whether I find more stuff that makes the same mistake.

TODO

- [x] ACE_movement inventory displayLoad.
- [x] ACE_Parachute respawn
- [x] ACE_Rearm respawn

inventory displayLoad causes the same problem. 70ms compiling a function everytime you open inventory.

### CODE REVIEW

File: addons/medical/CfgEventHandlers.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ class Extended_Init_EventHandlers {
 class Extended_Respawn_EventHandlers {
     class CAManBase {
         class ADDON {
-            respawn = QUOTE(call COMPILE_FILE(XEH_respawn));
+            respawn = QUOTE(call FUNC(handleRespawn));
```
Comment (commy2) [inline addons/medical/CfgEventHandlers.hpp]: CBA_fnc_addClassEventHandler both in preInit.
This would've avoided this problem in the first place.

Comment (dedmen) [reply addons/medical/CfgEventHandlers.hpp]: Should I also use addCEH for displayLoad?

**Comment by commy2** [comment]
lgtm

**Comment by dedmen** [comment]
Done.
Theoretically can do the same with all postInit handlers. They only run once.. But could improve mission start time?

**Comment by commy2** [comment]
Yes. If you do, separate PR please.

**Comment by dedmen** [comment]
I'll leave that to whoever has more time than me to do this :D

File: addons/medical/XEH_preInit.sqf [hunk 2]
```diff
@@ -41,6 +41,13 @@ private _fixStatic = {
 };
 ["StaticWeapon", "init", _fixStatic] call CBA_fnc_addClassEventHandler;
 ["Car", "init", _fixStatic] call CBA_fnc_addClassEventHandler;
+["CAManBase", "Init", FUNC(handleInit)] call CBA_fnc_addClassEventHandler;
+["CAManBase", "Respawn", FUNC(handleRespawn)] call CBA_fnc_addClassEventHandler;
+["CAManBase", "Killed", FUNC(handleKilled)] call CBA_fnc_addClassEventHandler;
+["CAManBase", "Local", FUNC(handleLocal)] call CBA_fnc_addClassEventHandler;
+
+["ACE_bodyBagObject", "init", EFUNC(dragging,initObject)] call CBA_fnc_addClassEventHandler;
```
Comment (PabstMirror) [inline addons/medical/XEH_preInit.sqf]: dragging does 
```
    class ThingX {
        class ADDON {
            init = QUOTE(_this call DFUNC(initObject));
```
and `"ACE_bodyBagObject" isKindOf "ThingX" = true`
so I think we can drop this

Comment (commy2) [reply addons/medical/XEH_preInit.sqf]: OK

Comment (dedmen) [reply addons/medical/XEH_preInit.sqf]: I didn't see a double init of dragging when profiling.. Maybe I just didn't look close enough though.

**Pull request merged**